packages/core/src/controllers/ConnectorController.ts

```diff
@@ -25,58 +30,73 @@ export const ConnectorController = {
     return subKey(state, key, callback)
   },
 
-  setConnectors(connectors: ConnectorControllerState['connectors'], multiChain?: boolean) {
-    if (multiChain) {
-      state.connectors = [...state.connectors, ...connectors.map(c => ref(c))]
+  setConnectors(connectors: ConnectorControllerState['connectors']) {
+    state.unMergedConnectors = [...state.unMergedConnectors, ...connectors]
+    state.connectors = this.mergeMultiChainConnectors(state.unMergedConnectors)
+  },
 
-      state.connectors = this.mergeMultiChainConnectors(state.connectors)
-    } else {
-      state.connectors = connectors.map(c => ref(c))
-    }
+  mergeMultiChainConnectors(connectors: Connector[]) {
+    const connectorsByNameMap = this.generateConnectorMapByName(connectors)
+
+    const refactoredConnectors = Array.from(connectorsByNameMap.values()).map(_connectors => {
+      if (_connectors.length > 1) {
+        return {
+          name: _connectors[0]?.name,
+          imageUrl: _connectors[0]?.imageUrl,
+          imageId: _connectors[0]?.imageId,
+          providers: this.getUniqueConnectorsByName(_connectors),
+          type: 'MULTI_CHAIN'
```

Comment:
Will it always be this type?

Reply:
It'll be this type oonly when there are multiple connectors with the same name